authentication_document.py

```diff
@@ -215,6 +216,60 @@ def from_dict(cls, _db, data, place_class=Place):
             place_class=place_class
         )
 
+    def update_library(self, library):
+        """Modify a library to reflect the current state of this
+        AuthenticationDocument.
+        
+        :param library: A Library.
+        :return: A ProblemDetail if there's a problem, otherwise None.
+        """
+        problem = self.update_audiences(library)
+        if not problem:
+            problem = self.update_service_areas(library)
+        return problem
+
+        
+    def update_audiences(self, library):
+        return self._update_audiences(library, self.audiences)
+
+    @classmethod
+    def _update_audiences(self, library, audiences):
+        original_audiences = audiences
+        if not audiences:
+            audiences = [Audience.PUBLIC]
+        if isinstance(audiences, basestring):
+            # This is invalid but we can easily support it.
+            audiences = [audiences]
+        if not isinstance(audiences, list):
+            return INVALID_AUTH_DOCUMENT.detailed(
+                _("'audience' must be a list") % audiences
+            )
+
+        # Ignore unrecognized audiences rather than rejecting the
+        # whole document.
+        audiences = [x for x in audiences
+                     if x in Audience.KNOWN_AUDIENCES]
+
+        # But there must be at least one audience we recognize.
+        if original_audiences and not audiences:
+            return INVALID_AUTH_DOCUMENT.detailed(
+                _("None of the provided audiences were recognized: %r") %
+                original_audiences
+            )
```

Comment:
It's for use by a library that wants to say they appeal to some other audience. Using it here is a good idea; my only concern is that if the library uses OTHER they're supposed to spell out who the audience is in their service_description, and if they think they've spelled it out in their `audience` they probably won't bother.